### PR TITLE
feat(carbon-provider): add deprecation warning for roundedCornersOptOut & focusRedesignOptOut

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Deprecates both feature flags and each respective flag's previous styling.

### Current behaviour

Currently, no deprecation warning exists for both feature flags. 

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/carbon-provider/carbon-provider.component.tsx [hunk 1]
```diff
@@ -33,6 +37,23 @@ export const CarbonProvider = ({
     existingRoundedCornersOptOut || roundedCornersOptOut;
   const focusRedesignOptOutValue =
     existingFocusRedesignOptOut || focusRedesignOptOut;
+
+  if (!deprecatedRoundedCornersOptOut && roundedCornersOptOutValue) {
+    deprecatedRoundedCornersOptOut = true;
+    Logger.deprecate(
+      "The `roundedCornersOptOut` feature flag has been deprecated and will soon be removed. " +
+        "Along with this feature flag, the previous pre rounded corners styling will also be removed. ",
```
Comment (DipperTheDan) [inline src/components/carbon-provider/carbon-provider.component.tsx:45]: **nitpick(non-blocking):** Grammarly reckons this should be hyphenated.

```suggestion
        "Along with this feature flag, the previous pre-rounded corners styling will also be removed. ",
```

File: src/components/carbon-provider/carbon-provider.component.tsx [hunk 2]
```diff
@@ -33,6 +37,23 @@ export const CarbonProvider = ({
...
+    );
+  }
+
+  if (!deprecatedFocusRedesignOptOut && focusRedesignOptOutValue) {
+    deprecatedFocusRedesignOptOut = true;
+    Logger.deprecate(
+      "The `focusRedesignOptOut` feature flag has been deprecated and will soon be removed. " +
+        "Along with this feature flag, the previous focus styling will also be removed. ",
```
Comment (edleeks87) [inline src/components/carbon-provider/carbon-provider.component.tsx:53]: same above wrt `legacy`

File: src/components/carbon-provider/__internal__/new-validation.context.ts [hunk 3]
```diff
@@ -5,10 +5,13 @@ export interface NewValidationContextProps {
    *
    * NOTE - Will eventually be set to `true` by default in the future. */
   validationRedesignOptIn?: boolean;
-  /** Feature flag for opting out of styling components to have rounded corners.
+  /** (Deprecated) Feature flag for opting out of styling components to have rounded corners.
    *
-   * NOTE - Will eventually be set to `false` by default in the future. */
+   * NOTE - This feature flag will soon be removed, along with the original styling. */
```
Comment (edleeks87) [inline src/components/carbon-provider/__internal__/new-validation.context.ts:10]:
```suggestion
   * NOTE - This feature flag will soon be removed, along with the legacy styling. */
```

File: src/components/carbon-provider/__internal__/new-validation.context.ts [hunk 4]
```diff
@@ -5,10 +5,13 @@ export interface NewValidationContextProps {
...
   roundedCornersOptOut?: boolean;
+  /** (Deprecated) Feature flag for opting out of the focus redesign.
+   *
+   * NOTE -  This feature flag will soon be removed, along with the original styling. */
```
Comment (edleeks87) [inline src/components/carbon-provider/__internal__/new-validation.context.ts:14]:
```suggestion
   * NOTE -  This feature flag will soon be removed, along with the legacy styling. */
```

**Review by DipperTheDan — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**